File: 6-kyu/take-a-ten-minute-walk/javascript/solution.js
function countOccurences(walk, direction)
{
  return walk.filter(dir => dir == direction).length;
}

function isValidWalk(walk) {
  let ns = 0;
  let ew = 0;
  if (walk.length == 10)
  {
    for (var pathIndex in walk)
    {
      if (countOccurences(walk, "w") == countOccurences(walk, "e") && countOccurences(walk, "n") == countOccurences(walk, "s"))
      {
        return true;
      }
      return false;
    }
  }
  return false;
}

File: 6-kyu/take-a-ten-minute-walk/ruby/solution.rb
def countOccurrences(walk, direction)
  counter = 0
  index = 0
  while walk[index]
    (walk[index] == direction) ? counter+=1 : nil
    index+=1
  end
  return counter
end

def return_true()
  return true
end

def is_valid_walk(walk)
  if (walk.length != 10)
    return false
  end
  return countOccurrences(walk, "w") == countOccurrences(walk, "e") && countOccurrences(walk, "n") == countOccurrences(walk, "s")
end